api/views.py

```diff
@@ -215,6 +218,39 @@ def get_queryset(self):
         return Comment.objects.get(pk=self.kwargs['comment_pk']).commentlike_set.all()
 
 
+def handle_inbox_request(request: Request, body: dict[str, Any]) -> Response:
+    from_user_id_url: str = body.get('actor').get('id')
+    to_user_id_url: str = body.get('object').get('id')
+
+    parsed_to_user_id = urlparse(to_user_id_url)
+    parsed_from_user_id = urlparse(from_user_id_url)
+
+    to_user_id = parsed_to_user_id.path.rsplit('/', 1)[-1]
+
+    try:
+        to_user = get_user_model().objects.get(id=to_user_id)
+    except get_user_model().DoesNotExist as e:
+        return Http404
+
+    if not parsed_from_user_id.hostname == request.get_host():
+        remote_request = RemoteRquest.objects.create(from_user_url=from_user_id_url, to_user=to_user)
+        remote_request.save()
+        return HttpResponse({}, status=status.HTTP_204_NO_CONTENT)
+
+    local_from_user_id = parsed_from_user_id.path.rsplit('/', 1)[-1]
+
+    try:
+        local_from_user = get_user_model().objects.get(id=local_from_user_id)
+    except get_user_model().DoesNotExist as e:
+        return Http404
+
+    request = Request.objects.create(from_user=local_from_user, to_user=to_user)
+    request.save()
+
+    serialized = RequestSerializer(request, context={'request': request}).data
+    return Response(serialized)
```

Comment:
We can also just return 204 since we don't really have an endpoint to get the request